sormas-ui/src/main/java/de/symeda/sormas/ui/location/LocationEditForm.java

```diff
@@ -524,19 +525,18 @@ private HorizontalLayout createGeoButton() {
 			e.getButton().removeStyleName(CssStyles.GEOCODE_BUTTON_HIGHLIGHT);
 		}, ValoTheme.BUTTON_ICON_ONLY, ValoTheme.BUTTON_BORDERLESS, ValoTheme.BUTTON_LARGE);
 
+		Field[] locationGeoFields = Stream.of(LocationDto.STREET, LocationDto.POSTAL_CODE, LocationDto.CITY, LocationDto.HOUSE_NUMBER)
+			.map(field -> (Field) getField(field))
+			.toArray(Field[]::new);
+
 		// Highlight geocode-button when the address changes
-		for (String fieldName : new String[] {
-			LocationDto.STREET,
-			LocationDto.POSTAL_CODE,
-			LocationDto.CITY,
-			LocationDto.HOUSE_NUMBER }) {
-			Field<?> field = getField(fieldName);
-			field.addValueChangeListener(e -> {
-				if (field.isModified()) {
-					geocodeButton.addStyleName(CssStyles.GEOCODE_BUTTON_HIGHLIGHT);
-				}
-			});
-		} ;
+		Stream.of(locationGeoFields).forEach(field -> field.addValueChangeListener(e -> {
+			if (isAllFieldsEmpty(locationGeoFields)) {
+				geocodeButton.removeStyleName(CssStyles.GEOCODE_BUTTON_HIGHLIGHT);
```

Comment:
When you empty all address fields, the geocoding button will not be highlighted. But will the coordinates already present be removed or stay there?

Reply:
Reviewed with @syntakker : Is okay like it is now, no functionality resets the coordinates yet if there are no useful coordinates.